opcua/server/address_space.py

```diff
@@ -233,6 +233,31 @@ def _add_node(self, item, user):
 
         return result
 
+    def delete_nodes(self, deletenodeitems, user=User.Admin):
+        results = []
+        for item in deletenodeitems:
+            results.append(self._delete_node(item, user))
+        return results
+
+    def _delete_node(self, item, user):
+        #TODO: Check if node is monitored if it is: "When any of the Nodes deleted by an invocation of this Service is being monitored, then a Notification containing the status code Bad_NodeIdUnknown is sent to the monitoring Client indicating that the Node has been deleted."
+        if not user == User.Admin:
+            return ua.StatusCode(ua.StatusCodes.BadUserAccessDenied)
+
+        if item.NodeId not in self._aspace:
+            self.logger.warning("DeleteNodesItem: node does not exists")
+            return ua.StatusCode(ua.StatusCodes.BadNodeIdUnknown)
+
+        if item.DeleteTargetReferences:
+            for elem in self._aspace.keys():
+                for rdesc in self._aspace[elem].references:
+                    if rdesc.NodeId == item.NodeId:
+                        self._aspace[elem].references.remove(rdesc)
+
+        del self._aspace[item.NodeId]
```

Comment:
Here instead of deleting it directly you can get the AttributeValueObject

    nodedata = self._aspace[item.NodeId]
    for cb in nodedata.attributes[AttributeId.Value].datachange_callbacks.value():
        try:
              cb(xxxxx, error=ua.StatusCode.UknownNodeId
        except:
             self.logger.exception("Error calling callbackxxxx")
     del self._aspace[item.NodeId]